v4-client-js/examples/websocket_orderbook_example.ts

```diff
@@ -0,0 +1,246 @@
+import { Network } from '../src/clients/constants';
+import { SocketClient } from '../src/clients/socket-client';
+
+function test(): void {
+    let orderBookBidList: [number, number, number][] = [];
+    let orderBookAskList: [number, number, number][] = [];
+
+    const mySocket = new SocketClient(
+        Network.mainnet().indexerConfig,
+        () => {
+            console.log('socket opened');
+
+            mySocket.subscribeToOrderbook('ETH-USD');
+        },
+        () => {
+            console.log('socket closed');
+        },
+        (message) => {
+            try {
+                if (typeof message.data === "string") {
+                    const jsonString = message.data as string;
+                    const parsingData = JSON.parse(jsonString);
+                    const messageId = parsingData.message_id;
+                    const orderBookDataList = parsingData.contents;
+
+                    if (orderBookDataList instanceof Array) {
+                        // common orderBook data;
+                        [orderBookBidList, orderBookAskList] = updateOrderBook(
+                            orderBookDataList,
+                            orderBookBidList,
+                            orderBookAskList,
+                            messageId,
+                        );
+
+                        // sort
+                        orderBookBidList = sortByNthElementDesc(
+                            orderBookBidList,
+                            0,
+                        );
+                        orderBookAskList = sortByNthElementAsc(
+                            orderBookAskList,
+                            0,
+                        );
+
+                        // resolving crossed orderBook
+                        if (orderBookBidList[0][0] >= orderBookAskList[0][0]) {
```

Comment:
needs additional checks to ensure that both `orderBookBidList` and `orderBookAskList` is not undefined and has a length > 0.

On very rare occasions, a market may only have one side of the book.